gnocchi/storage/incoming/redis.py

```diff
@@ -49,13 +48,27 @@ def _store_new_measures(self, metric, data):
         self._client.rpush(path, data)
 
     def _build_report(self, details):
+        build_metric_list = (lambda x: x.decode('utf8').split(redis.SEP)[1]
+                             if details else lambda x: True)
+        metrics = 0
+        measures = 0
         match = redis.SEP.join([self.get_sack_name("*"), "*"])
-        metric_details = collections.defaultdict(int)
-        for key in self._client.scan_iter(match=match, count=1000):
-            metric = key.decode('utf8').split(redis.SEP)[1]
-            metric_details[metric] = self._client.llen(key)
-        return (len(metric_details.keys()), sum(metric_details.values()),
-                metric_details if details else None)
+        metric_details = {}
+        cursor = '0'
+        while cursor != 0:
```

Comment:
i wanted to leverage pipeline and i don't want to pipe 1 million commands to redis and then get a million item list in return. also, didn't want to add a check every loop to decide whether to chunk or not, although i guess this relates back to other patch and can test how much a million single if checks adds to workload